nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -117,6 +129,86 @@ public static DocumentModel retrieveContentFromColdStorage(CoreSession session,
         return session.saveDocument(documentModel);
     }
 
+    /**
+     * Checks if the retrieved cold storage contents are available for download.
+     *
+     * @implSpec: Queries all documents with a cold storage content which are being retrieved, meaning
+     *            {@value COLD_STORAGE_BEING_RETRIEVED_PROPERTY} is {@code true}, and it checks if it is available for
+     *            download. In which case its fires a {@value COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME} event.
+     * @see #retrieveContentFromColdStorage(CoreSession, DocumentRef, int)
+     */
+    // TODO once the cold storage content is available for download (wait for XP-28417 to be done), we should store the
+    // retrieved numberOfDaysOfAvailability in cold storage schema
```

Comment:
Not what would be the point of storing numberOfDaysOfAvailability or the effective end date, because we don't know it exactly (it's advisory to S3, and we don't know precisely the timezone mechanics used (or they could change)). What usage would there be for it?
There will be a blob status query to check restore / availability that will return this info from S3 anyway.

Reply:
This was only an assumption we made, not knowing how the low-level implementation would handle it. Good if it's not needed.

Reply:
i will remove the unneeded javadoc comments once we will have the low level implementation